src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java

```diff
@@ -30,123 +34,158 @@
-            if (null != projectView) {
-                try {
-                    blackDuckApiClient.delete(projectView);
-                } catch (Exception ignored) {
-                    // ignored
-                }
-            }
-            if (null != userView) {
-                try {
-                    blackDuckApiClient.delete(userView);
-                } catch (Exception ignored) {
-                    // ignored
-                }
+            cleanView(projectView);
+            cleanView(userView);
+        }
+    }
+
+    @Test
+    public void testInvalidUser() throws IntegrationException {
+        String projectName = "user-group-project" + System.currentTimeMillis();
+        String userName = "user" + System.currentTimeMillis();
+        ProjectView projectView = null;
+
+        try {
+            ProjectSyncModel projectSyncModel = ProjectSyncModel.createWithDefaults(projectName, "test");
+            PROJECT_SERVICE.syncProjectAndVersion(projectSyncModel);
+
+            projectView = PROJECT_SERVICE.getProjectByName(projectName).orElse(null);
+
+            ProjectView finalProjectView = projectView;
+            assertThrows(BlackDuckIntegrationException.class, () -> PROJECT_USERS_SERVICE.addUserToProject(finalProjectView, userName));
+        } finally {
+            cleanView(projectView);
+        }
+    }
+
+    @Test
+    public void testGetActiveUsersForProject() throws IntegrationException {
+        String projectName = "user-group-project" + System.currentTimeMillis();
+        String inActiveUser = "inactive-user" + System.currentTimeMillis();
+        String activeUser = "active-user" + System.currentTimeMillis();
+        ProjectView projectView = null;
+        UserView inActiveUserView = null;
+        UserView activeUserView = null;
+        Set<UserView> allActiveUsersForProject = null;
+
+        try {
+            ProjectSyncModel projectSyncModel = ProjectSyncModel.createWithDefaults(projectName, "test");
+            PROJECT_SERVICE.syncProjectAndVersion(projectSyncModel);
+            projectView = PROJECT_SERVICE.getProjectByName(projectName).orElse(null);
+
+            inActiveUserView = createUserView(inActiveUser, false);
+            activeUserView = createUserView(activeUser, true);
+
+            allActiveUsersForProject = PROJECT_USERS_SERVICE.getAllActiveUsersForProject(projectView);
+            assertFalse(allActiveUsersForProject.contains(inActiveUserView));
+            assertFalse(allActiveUsersForProject.contains(activeUserView));
+
+            PROJECT_USERS_SERVICE.addUserToProject(projectView, inActiveUser);
+            PROJECT_USERS_SERVICE.addUserToProject(projectView, activeUser);
+
+            allActiveUsersForProject = PROJECT_USERS_SERVICE.getAllActiveUsersForProject(projectView);
+            assertFalse(allActiveUsersForProject.contains(inActiveUserView));
+            assertTrue(allActiveUsersForProject.contains(activeUserView));
+        } finally {
```

Comment:
Given that we've removed the check for existing users, we could use this current stone to kill a few more birds and try adding the users AGAIN and verify that there are no exceptions and the returned list doesn't change in terms of size or content.